apps/keyboard/test/marionette/number_keyboard_test.js

```diff
@@ -0,0 +1,164 @@
+/* global suite */
+
+'use strict';
+
+var KeyboardTestApp = require('./lib/keyboard_test_app'),
+    Keyboard = require('./lib/keyboard'),
+    assert = require('assert');
+
+marionette('Number keyboard input tests', function() {
+  var apps = {};
+  var keyboardTestApp = null;
+  var systemInputMgmt = null;
+  var keyboard = null;
+  var client = null;
+  var typeGroupMap = {
+    'text': 'text',
+    'textarea': 'text',
+    'url': 'url',
+    'email': 'email',
+    'password': 'password',
+    'search': 'text',
+    'number': 'number',
+    'tel': 'number'
+  };
+
+  apps[KeyboardTestApp.ORIGIN] = __dirname + '/keyboardtestapp';
+
+  client = marionette.client({
+    profile: {
+      apps: apps,
+      prefs: {
+        'focusmanager.testmode': true
+      },
+      settings: {
+        'lockscreen.enabled': false,
+        'ftu.manifestURL': null
+      }
+    }
+  });
+
+  setup(function() {
+    keyboard =  new Keyboard(client);
+    systemInputMgmt = client.loader.getAppClass('system', 'input_management');
+
+    // create a keyboard test app
+    keyboardTestApp = new KeyboardTestApp(client);
+    keyboardTestApp.launch();
+  });
+
+  suite('input type tests', function() {
+    setup(function() {
+      // Switch to test app frame.
+      client.switchToFrame();
+      client.apps.switchToApp(KeyboardTestApp.ORIGIN);
+    });
+
+    test('number input', function() {
+      keyboardTestApp.numberInput.tap();
+
+      systemInputMgmt.waitForKeyboardFrameDisplayed();
+      systemInputMgmt.switchToActiveKeyboardFrame();
+
+      assert.equal(
+        keyboard.getCurrentInputType(), typeGroupMap.number);
```

Comment:
Please add another assertion here proving that the current active layout has no alphabet keys.